forEach/challenges-01.test.js

```diff
@@ -0,0 +1,356 @@
+"use strict";
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 1
+//
+// Write a function named pushIntoSecond which takes in an array and uses a for loop
+// to push all of the elements from the initial array into a second array.
+//
+// Return the second array.
+// ------------------------------------------------------------------------------------------------
+
+const pushIntoSecond = arr => {
+  const second = [];
+  // Solution code here...
+  for (let i = 0, l = arr.length; i < l; i++) {
+    second[i] = arr[i];
+  }
+  return second;
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 2
+//
+// Write a function named addBonusPoints that takes in an array of raw test scores
+// and adds five bonus points to each score.
+//
+// Use the for...of syntax.
+//
+// Return an array of scores that have had the bonus points added.
+// ------------------------------------------------------------------------------------------------
+
+const addBonusPoints = arr => {
+  // Solution code here...
+  let arrResult = [];
+  for (let elm of arr) {
+    arrResult.push(elm + 5);
+  }
+  return arrResult;
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 3
+//
+// Write a function named addCurve that takes in an array of raw test scores and increases each score by 5%.
+//
+// Use the for...in syntax.
+//
+// Return an array of curved scores;
+// ------------------------------------------------------------------------------------------------
+const addCurve = arr => {
+  // Solution code here...
+  var arrResult = [];
+  for (let i in arr) {
+    let result = arr[i] * 1.05;
+    arrResult.push(result);
+  }
+  return arrResult;
+};
+
```

Comment:
when using for...in you don't need to make a new result array, it will update the array itself. So you could shorten this code down to `const addCurve = (arr) => {
  for (let i in arr) {
    arr[i] *= 1.05;
  }
  return arr;
}`